machine-runner/src/design/protocol.ts

```diff
@@ -202,19 +180,19 @@ namespace ImplMachine {
   export const make = <
     SwarmProtocolName extends string,
     MachineName extends string,
-    RegisteredEventsFactoriesTuple extends Readonly<MachineEvent.Factory.Any[]>,
+    MachineEventFactories extends MachineEvent.Factory.Any,
   >(
     swarmName: SwarmProtocolName,
     machineName: MachineName,
-    registeredEventFactories: RegisteredEventsFactoriesTuple,
-  ): Machine<SwarmProtocolName, MachineName, RegisteredEventsFactoriesTuple> => {
-    type Self = Machine<SwarmProtocolName, MachineName, RegisteredEventsFactoriesTuple>
-    type Protocol = MachineProtocol<SwarmProtocolName, MachineName, RegisteredEventsFactoriesTuple>
+    registeredEventFactories: MachineEventFactories[],
+  ): Machine<SwarmProtocolName, MachineName, MachineEventFactories> => {
+    type Self = Machine<SwarmProtocolName, MachineName, MachineEventFactories>
+    type Protocol = MachineProtocol<SwarmProtocolName, MachineName, MachineEventFactories>
 
     const protocol: Protocol = {
       swarmName: swarmName,
       name: machineName,
-      registeredEvents: registeredEventFactories,
+      registeredEvents: [...registeredEventFactories],
```

Comment:
👍 (should have a comment)

Reply:
This line is no longer necessary

Reply:
fixed in https://github.com/Actyx/machines/pull/41/commits/a1f2e1537b83ed1ba115bb33aba3bc1136eaa849

Reply:
ah, didn’t we want to copy the array in case the user modifies it later?

Reply:
Line 62 will be sufficient for that. I'll add comment on that line instead